### PR TITLE
refactor!: remove deprecated GeneratedVaadinNotification

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

1. Removed deprecated `GeneratedVaadinNotification` class,
2. Updated `Notification` to implement `HasThemeVariant`.

## Type of change

- Breaking change

### CODE REVIEW

File: vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java [hunk 1]
```diff
@@ -18,21 +18,26 @@
...
 import com.vaadin.flow.component.AttachEvent;
 import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.ComponentEvent;
 import com.vaadin.flow.component.ComponentEventListener;
 import com.vaadin.flow.component.ComponentUtil;
 import com.vaadin.flow.component.DetachEvent;
 import com.vaadin.flow.component.HasComponents;
+import com.vaadin.flow.component.HasSize;
 import com.vaadin.flow.component.HasStyle;
 import com.vaadin.flow.component.HasTheme;
```
Comment (sissbruecker) [inline vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java:33]: `HasSize`, `HasTheme` are unused.

File: vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java [hunk 2]
```diff
@@ -464,52 +474,66 @@ && getElement().getNode().getParent() == null) {
...
      * The duration in milliseconds to show the notification. Set to {@code 0}
      * or a negative number to disable the notification auto-closing.
-     * </p>
      *
      * @param duration
      *            the value to set
      */
     public void setDuration(int duration) {
-        setDuration((double) duration);
+        getElement().setProperty("duration", (double) duration);
```
Comment (sissbruecker) [inline vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java:533]:
```suggestion
        getElement().setProperty("duration", duration);
```
Nit: Cast is redundant, there is an overload for ints.

File: vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java [hunk 3]
```diff
@@ -521,7 +545,7 @@ public void setDuration(int duration) {
      * @return the {@code duration} property from the webcomponent
      */
     public int getDuration() {
-        return (int) getDurationDouble();
+        return (int) getElement().getProperty("duration", 0.0);
```
Comment (sissbruecker) [inline vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java:548]:
```suggestion
        return getElement().getProperty("duration", 0);
```
Nit: Cast is redundant, there is an overload for ints.

File: vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java (hunk 1 above)
Comment (web-padawan) [reply vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java]: Removed.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**